src/daemons/rhsmcertd.c

```diff
@@ -425,13 +438,25 @@ get_bool_from_config_file (GKeyFile * key_file, const char *group, const char *k
 {
     GError *error = NULL;
     bool value = g_key_file_get_boolean (key_file, group, key, &error);
-        // If key does not exist in config file, return the default_value given
+    // If key does not exist in config file, return the default_value given
     if (error != NULL && (error->code == G_KEY_FILE_ERROR_KEY_NOT_FOUND || error->code == G_KEY_FILE_ERROR_INVALID_VALUE)) {
+        debug ("Key %s does not exists in the group %s. Using default value: %d", key, group, default_value);
         value = default_value;
     }
     return value;
 }
 
+gchar *
+get_string_from_config_file (GKeyFile * key_file, const char *group, const char *key)
+{
+    GError *error = NULL;
+    char *value = g_key_file_get_string (key_file, group, key, &error);
+    if (error != NULL && (error->code == G_KEY_FILE_ERROR_KEY_NOT_FOUND || error->code == G_KEY_FILE_ERROR_INVALID_VALUE)) {
```

Comment:
reading the api documentation for `g_key_file_get_string`: it does not list `G_KEY_FILE_ERROR_INVALID_VALUE` as possible error on failure; that error seems specific to functions that not only read a value, but also interpret it (e.g. `g_key_file_get_integer`, as it tries to parse the value as integer); hence, i'd remove the check for `G_KEY_FILE_ERROR_INVALID_VALUE` here, as it should not happen